tests/python/unittest/test_meta_schedule_multi_anchor.py

```diff
@@ -0,0 +1,131 @@
+# specific language governing permissions and limitations
+# under the License.
+import os
+import tempfile
+
+import numpy as np
+
+import tvm
+import tvm.testing
+from tvm import relay
+from tvm.meta_schedule.tune import Parse, extract_task_from_relay
+from tvm.meta_schedule.database import TuningRecord, JSONDatabase
+from tvm.meta_schedule.integration import ApplyHistoryBest
+
+
+def get_dense_dense(data_shape, weight_shape):
+    def multi_dense():
+        p_data = relay.var("p_data", shape=data_shape, dtype="float32")
+        p_weight1 = relay.var("p_weight1", shape=weight_shape, dtype="float32")
+        p_weight2 = relay.var("p_weight2", shape=weight_shape, dtype="float32")
+
+        dense1 = relay.nn.dense(p_data, p_weight1)
+        dense2 = relay.nn.dense(dense1, p_weight2)
+
+        f = relay.Function([p_data, p_weight1, p_weight2], dense2)
+        f = f.with_attr("Primitive", tvm.tir.IntImm("int32", 1))
+        return f
+
+    data = relay.var("data", shape=data_shape, dtype="float32")
+    weight1 = relay.var("weight1", shape=weight_shape, dtype="float32")
+    weight2 = relay.var("weight2", shape=weight_shape, dtype="float32")
+
+    out = relay.Call(multi_dense(), [data, weight1, weight2])
+    return relay.Function([data, weight1, weight2], out)
+
+
+def get_ref(data_np, weight1_np, weight2_np):
+    dense1 = np.dot(data_np, np.transpose(weight1_np))
+    return np.dot(dense1, np.transpose(weight2_np))
+
+
+def schedule_dense_dense(sch):
+    dense1 = sch.get_block("T_matmul_NT")
+    dense2 = sch.get_block("T_matmul_NT_1")
+
+    y1, x1, k1 = sch.get_loops(dense1)
+    y2, x2, k2 = sch.get_loops(dense2)
+
+    # ...
+
+
+def test_dense_dense():
+    M, N, K = 128, 128, 128
+    data_shape = (M, K)
+    weight_shape = (N, K)
+
+    relay_mod = tvm.IRModule.from_expr(get_dense_dense(data_shape, weight_shape))
+
+    # print(relay.transform.InferType()(relay_mod))
+
+    target = "llvm"
+
+    data_np = np.random.randn(*data_shape).astype("float32")
+    weight1_np = np.random.randn(*weight_shape).astype("float32")
+    weight2_np = np.random.randn(*weight_shape).astype("float32")
+
+    params = {"weight1": weight1_np, "weight2": weight2_np}
+
+    extracted_tasks = extract_task_from_relay(relay_mod, target, params)
+
+    assert len(extracted_tasks) == 1
+
+    task = extracted_tasks[0]
+
+    mod = Parse._mod(task.dispatched[0])
+
+    with tempfile.TemporaryDirectory() as work_dir:
+        database = JSONDatabase(
+            path_workload=os.path.join(work_dir, "database_workload.json"),
+            path_tuning_record=os.path.join(work_dir, "database_tuning_record.json"),
+        )
+
+        workload = database.commit_workload(mod)
+
+        sch = tvm.tir.Schedule(mod)
+
+        schedule_dense_dense(sch)
+
+        # print(sch.mod.script())
+
+        tune_rec = TuningRecord(sch.trace, [0.0], workload, tvm.target.Target(target), [])
+
+        database.commit_tuning_record(tune_rec)
```

Comment:
@junrushao1994 @zxybazh 

I keep writing this database boilerplate for manual scheduling. I'm thinking about a clean API so that users don't have to go through the explicit task extraction -> database creation steps. Right now it looks like

```
    relay_mod = tvm.IRModule.from_expr(...)
    target = "llvm"    
    params = {"weight1": weight1_np, "weight2": weight2_np}

    def schedule_fn(task, sch):
        if "nn_dense_nn_dense" in task.task_name:
            schedule_dense_dense(sch)
            return True
        return False

    database = apply_manual_schedules(relay_mod, target, params, schedule_fn)

    with ApplyHistoryBest(database):
           ...
```

If this looks ok, I can PR it after this one.

Reply:
There is a DummyDatabase class in `python/tvm/meta_schedule/testing/utils.py` where we don't need to create json files for intermediate results, and I was wondering if we could further reduce boilerplate by enhancing that class. What do you think?

Reply:
Yeah using the `Dummy` classes can be very helpful in tuning and it's acutally a good idea to provide new use interface as you mentioned, a schedule function to work on on the relay level. Let me know when you got the PR ready : )